### PR TITLE
Fix various bugs and issues in crypto/elytron

### LINKED ISSUE #23173: crypto/elytron package has several bugs
### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the [latest nightly release](https://github.com/keycloak/keycloak/releases/tag/nightly)

### Area

core

### Describe the bug

The crypto/elytron package was created as a potential option to supporting crypto functions without the need to hardcode dependencies to either BC or BCFIPS. This would allow the user to configure their preferred crypto provider. But, due to limitations (mostly with the OpenJDK FIPS provider) , this package is not ready for use. That said, as the dev of this package, I have done some additional testing and found that a few issues that I need to fix. A key issue is a  failure to provide a complete and thorough implementation parsing DER elements.

### Version

latest

### Expected behavior

DER elements from certs are parsed correctly.

### Actual behavior

The DER element parsing was either wrong, or specific to a given test cert and unable to handle other variations.

### How to Reproduce?

Tested by switching keycloak to use the crytpo/elytron package instead of BC(FIPS). And running unit and integration tests.

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by david-rh**
Closes #23173 

I did some additional testing of this module, and discovered a few bugs, and sections where processing of DER elements was implemented wrong.

### CODE REVIEW

File: crypto/elytron/src/main/java/org/keycloak/crypto/elytron/ElytronCertificateUtils.java [hunk 1]
```diff
@@ -249,16 +252,26 @@ public List<String> getCRLDistributionPoints(X509Certificate cert) throws IOExce
...
+                   der.decodeImplicit(0x86);
+                   distPointUrls.add(der.decodeOctetStringAsString());
+                   log.debug("Adding Dist point name: " + distPointUrls.get(distPointUrls.size()-1));
                    break;
                 default:
                    der.skipElement();
+            }
+            // Check to see if there is another sequence to process
+            try {
+                if(!der.hasNextElement() && der.peekType() == ASN1.SEQUENCE_TYPE) {
```
Comment (pedroigor) [inline crypto/elytron/src/main/java/org/keycloak/crypto/elytron/ElytronCertificateUtils.java:268]: Would mind explaining why we are expecting exceptions and ignoring them here?

Not sure if I'm following the call to decode a sequence if there are no elements to process?

**Review by pedroigor — COMMENTED** [review]
I appreciate if you create an issue first with all the issues you are fixing to make it easier to map them with the changes you are doing.

**Comment by mposolda** [comment]
@david-rh Thanks for the PR!

+1 for creating the GH issue(s) of what is this fixing. It can be also good to add them to the FIPS project for tracking purposes https://github.com/orgs/keycloak/projects/13 .

File: crypto/elytron/src/main/java/org/keycloak/crypto/elytron/ElytronCertificateUtils.java (hunk 1 above)
Comment (david-rh) [reply crypto/elytron/src/main/java/org/keycloak/crypto/elytron/ElytronCertificateUtils.java:268]: I added a log to at least print out the exception. As I was testing, I would run into an issue where this check for additional elements would throw an exception when there weren't any more elements. I eventually figured out what I was doing wrong in those cases. But, I left the try catch block to handle it if other cert encodings cause the same issue.

**Review by mposolda — APPROVED** [review]

**Comment by mposolda** [comment]
@david-rh Thanks!

**Pull request merged**